app/Http/Controllers/EventsController.php

```diff
@@ -123,6 +123,9 @@ public function edit(Event $event)
 	public function update(Event $event, Requests\EventRequest $request)
 	{
 		$this->authorize("update", $event);
+		$request->merge([
+			'cancelled_at' => ($request->cancelled) ? date('c') : null
```

Comment:
Add the cancelled_at logic in the entity by using a `setCancelledAttribute` and use `Carbon::now()` instead of native date functions.

Reply:
I'm not sure how you would do that. A mutator only receives the value, not the entire request:

```
public function setFirstNameAttribute($value)
    {
        $this->attributes['first_name'] = strtolower($value);
    }
```

Here I want to add a new attribute to the request, so I can't use a mutator. Or perhaps if I call the form element `cancelled_at`, but that would make it difficult to display the right boolean value.

Reply:
No, wait. Now that I'm actually looking at the example I just posted, I think it is possible 😅  still needed to wake up I guess.

Reply:
OK, it does **not** seem to work. The mutator is not called when doing `$event->update($request->all())` in the controller. Maybe because there is no column `cancelled` in the database. Or maybe because of something with eloquent vs query builder as seen here: https://stackoverflow.com/questions/51096280/laravel-eloquent-mutators-not-work-on-update-data

Reply:
@blackshadev can you help? I've pushed the latest code.